server/master/src/main/java/org/apache/accumulo/master/tableOps/tableImport/PopulateMetadataTable.java

```diff
@@ -87,110 +86,112 @@
   @Override
   public Repo<Master> call(long tid, Master master) throws Exception {
 
-    Path path = new Path(tableInfo.exportDir, Constants.EXPORT_FILE);
+    Path path = new Path(tableInfo.exportFile);
 
     BatchWriter mbw = null;
     ZipInputStream zis = null;
 
     try {
       VolumeManager fs = master.getFileSystem();
+      final String[] volumes = ServerConstants.getBaseUris(master.getContext());
 
       mbw = master.getContext().createBatchWriter(MetadataTable.NAME, new BatchWriterConfig());
 
       zis = new ZipInputStream(fs.open(path));
```

Comment:
It seems like this `ZipInputStream` should be created inside the loop over `tableInfo.directories`, but I am not sure. Will `zipEntry.getName().equals(Constants.EXPORT_METADATA_FILE)` only be true for the first directory mapping and false for all subsequent directory mappings? Depends on the behavior of `zis.getNextEntry()`, which I suspect only iterates through the zip file entries once.

Reply:
There is one export file, and thus one `EXPORT_METADATA_FILE` despite the number of `tableInfo.directories`. It is expected that the sets of rfiles referenced in both places is congruent, but we could positively check that this is the case in this code.

I'm definitely missing some code here. We iterate over the original metadata table entries, I haven't tracked which member from `tableInfo.directories` each original file was found in and thus end up failing the `exists` check in `MoveExportedFiles`. I need to go back to the drawing board here and change things a bit so that information can be included in the `fileNameMappings` structure.